src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,124 @@
+     * @throws NumberFormatException if user doesn't enter valid integers when entering task numbers.
+     */
+    public static void parse(String userInput, DukeTaskList dukeTaskList) {
+        String[] slashSplitInput = userInput.split("/", 2);
+        String[] spaceSplitInput = slashSplitInput[0].split(" ", 2); // Everything before the slash, split again by space
+
+        try {
+            if (userInput.equals("bye")) { // Exit
+                Ui.printFarewell();
+                isRunning = false;
+            } else if (userInput.equals("list")) {
+                Ui.printList(dukeTaskList.sendListForPrint());
+            } else if (spaceSplitInput[0].equals("done")) {
+                if (spaceSplitInput.length < 2) {
+                    // No task number entered
+                    throw new DukeException("Done Task number is missing!\n");
+                }
+                int doneTaskNo = Integer.parseInt(spaceSplitInput[1]); // Throws NumberFormatException if string cannot be parsed into valid int
+
+                dukeTaskList.doneTask(doneTaskNo);
+            } else if (spaceSplitInput[0].equals("delete")) {
+                if (spaceSplitInput.length < 2) {
+                    // No task number entered
+                    throw new DukeException("Delete task number is missing!\n");
+                }
+                int deleteTaskNo = Integer.parseInt(spaceSplitInput[1]); // Throws NumberFormatException if string cannot be parsed into valid int
+
+                dukeTaskList.deleteTask(deleteTaskNo);
+            } else if (spaceSplitInput[0].equals("find")) {
+                if (spaceSplitInput.length < 2) {
+                    // No find keyword entered
+                    throw new DukeException("Find keyword is missing!\n");
+                }
+                String keyword = spaceSplitInput[1];
+
+                dukeTaskList.searchTask(keyword);
+            } else if (spaceSplitInput[0].equals("todo")) {
+                if (spaceSplitInput.length < 2) {
+                    // Tudo has no description. If has, spaceSplitInput has length 2.
+                    throw new DukeException("Todo description cannot be empty!\n");
+                }
+                String toDoText = spaceSplitInput[1].trim();
+                dukeTaskList.addToDo(toDoText);
+            } else if (spaceSplitInput[0].equals("deadline")) {
+                if (spaceSplitInput.length < 2) {
+                    // Deadline has no description
+                    throw new DukeException("Deadline description cannot be empty!\n");
+                }
+                if (slashSplitInput.length < 2) {
+                    // Deadline has no slash (thus no date)
+                    throw new DukeException("Deadline must have a date and time written after a slash!\n");
+                }
+
+                String[] DdlDateTimeArr = slashSplitInput[1].split(" ", 3);
+                if (DdlDateTimeArr.length < 3) {
```

Comment:
Perhaps you could abstract out 3 into a final variable in case the format of input changes.